### ISSUE TITLE
Cordova Uncaught (in promise) Error creating peer connection Promise.then (async)

### ISSUE THREAD

**Issue by CreativeWolf** [feross/simple-peer#459]
Heya,

Here's the code used:
```
<html>
    <head>
        <title>Test</title>
        <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no" />

<script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>


    </head>
<body >
<div data-role="page">

<div role="main" class="ui-content" data-inset="false">
<fieldset data-role="controlgroup" data-type="horizontal" data-mini="true">
  <legend>Radio Buttons</legend>
  <input type="radio" name="role" id="radio-choice-ab" class="role" value="1" >
  <label for="radio-choice-ab">Faculty</label>
  <input type="radio" name="role" id="radio-choice-bb" class="role" value="2">
  <label for="radio-choice-bb">Student</label>
  
</fieldset>
       <form id="idGenarate">
      <textarea id="incoming"></textarea>
      <button type="submit">submit</button>
    </form>
    <div id="connectionCode">
    </div>

        <div id="messageReceived">
        
        </div>

    <form id="sendMsgForm">
        <input type="text" name="" id="messageText">
        <button type="submit">Send Message</button>
    </form>
     <pre id="outgoing"></pre>

            </div>

        </div>
 
        <script src="https://code.jquery.com/ui/1.11.4/jquery-ui.min.js"></script>
        <script src="https://ajax.googleapis.com/ajax/libs/jquerymobile/1.4.5/jquery.mobile.min.js"></script>
        <script type="text/javascript" src="js/simplepeer.min.js"></script>
        <script type="text/javascript" src="js/main.js"></script>
        <script src="cordova.js"></script>
      
          
    </body>
</html>
```
```
function testRtc(){
var p = new SimplePeer({ initiator: location.hash === '#1', trickle: false })
p.on('error', err => console.log('error', err))
p.on('signal', data => {
  console.log('SIGNAL', JSON.stringify(data))
  document.querySelector('#outgoing').textContent = JSON.stringify(data)
})

document.querySelector('form').addEventListener('submit', ev => {
  ev.preventDefault()
  p.signal(JSON.parse(document.querySelector('#incoming').value))
})

p.on('connect', () => {
  console.log('CONNECT')
  p.send('whatever' + Math.random())
})

p.on('data', data => {
  console.log('data: ' + data)
})
}
```
When tested in browser it works fine, the function testRtc() is called successfully, but in cordova, it's throwing the following errors while trying to instantiate SimplePeer using :

`new SimplePeer({ initiator: true });`

> cordova.js:309 Error in Error callbackId: WebRTCPlugin279177558 : Error creating peer connection
> r {_id: "6647244", _readableState: d, readable: true, _events: {…}, _eventsCount: 2, …}allowHalfOpen: falseallowHalfTrickle: falseanswerOptions: {}channelConfig: {}channelName: undefinedconfig: {iceServers: Array(2), sdpSemantics: "unified-plan"}connected: falseiceCompleteTimeout: 5000initiator: truelocalAddress: undefinedlocalFamily: undefinedlocalPort: undefinedofferOptions: {}readable: trueremoteAddress: undefinedremoteFamily: undefinedremotePort: undefinedsdpTransform: ƒ (e)streams: []trickle: truewritable: true_batchedNegotiation: false_cb: null_channel: InnerRTCDataChannel {binaryType: "arraybuffer", onmessage: ƒ, onbufferedamountlow: ƒ, onopen: ƒ, onclose: ƒ, …}_channelReady: false_chunk: null_closingInterval: 21_events: {end: ƒ, finish: ƒ}_eventsCount: 2_firstStable: true_iceComplete: false_iceCompleteTimer: null_id: "6647244"_interval: null_isNegotiating: true_isReactNativeWebrtc: false_maxListeners: undefined_onFinishBound: ƒ ()_pc: RTCPeerConnection {id: 2, iceConnectionState: "new", iceGatheringState: "new", signalingState: "stable", oniceconnectionstatechange: ƒ, …}_pcReady: false_pendingCandidates: []_queuedNegotiation: false_readableState: d {objectMode: false, highWaterMark: 16384, buffer: e, length: 0, pipes: null, …}_remoteStreams: []_remoteTracks: []_senderMap: Map(0) {}_sendersAwaitingStable: []_writableState: l {objectMode: false, highWaterMark: 16384, finalCalled: false, needDrain: false, ending: false, …}_wrtc: {RTCPeerConnection: ƒ, RTCSessionDescription: ƒ, RTCIceCandidate: ƒ}bufferSize: (...)destroyed: (...)__proto__: n

> index.html:1 Uncaught (in promise) Error creating peer connection
> Promise.then (async)
> nextTick @ cordova.js:931
> androidExec @ cordova.js:973
> RTCPeerConnection @ webrtc.js:480
> r @ simplepeer.min.js:11
> (anonymous) @ VM92:1

> simplepeer.min.js:11 Uncaught TypeError: Cannot read property 'then' of undefined
>     at r._createOffer (simplepeer.min.js:11)
>     at simplepeer.min.js:11

However, the same also works when used outside the function testRtc()

Appreciate any help on this please.

**Comment by t-mullen** [comment]
What are you using to add WebRTC to Cordova?

**Comment by CreativeWolf** [comment]
Nothing additional - It does work right out of box.
Instantiation works on load and anytime after that scope is lost.
Been successful at creating two different apps and establishing connection due to this limitation.
Happy to share any other additional info.

**Comment by CreativeWolf** [comment]
Heya,
This is to update that issue stands resolved. After some debugging, found that I had indeed installed a plugin webrtc for cordova and that was messing with the scope. Removed it and now it's working as intended and I'm able to instantiate, pass the offer, answer across using Laravel (for now), establish connection, send data between two peers using a single app.

Now trying to tinker with using ejabberd + jingle as signalling server instead of Laravel (as ejabbered is already used in the application for chat).

Thanks!
